core/templates/dev/head/pages/exploration-player-page/suggestion-modal-for-learner-local-view/suggestion-modal-for-exploration-player.service.ts

```diff
@@ -61,10 +63,29 @@ angular.module('oppia').factory('SuggestionModalForExplorationPlayerService', [
             $timeout(function() {
               $scope.showEditor = true;
             }, 500);
-
             $scope.cancelSuggestion = function() {
               SuggestionModalService.cancelSuggestion($uibModalInstance);
             };
+            $scope.$on('changed', function(evt, data) {
+              var compare2 = data;
+              var compare = $scope.originalHtml;
+              var undoRegExp = new RegExp(['<div class="oppia-rte-component',
```

Comment:
This is all a bit opaque to me. Perhaps you can explain a bit more clearly what you're doing?

The variable names should be clearer, too. "compare" is a verb, and "compare2" isn't clear. Try to give them more semantic names that describe what they mean.

Reply:
Here I'm trying to undo the changes done by the `wrapComponents` function in `ck-editor-4-rte.directive.ts.,`
i.e remove the div tags added by the function and compare with the original html.
I'll give the variables more meaningful names.